sdk/core/Azure.Core/samples/DynamicContent.md

```diff
@@ -0,0 +1,152 @@
+# Azure SDK Dynamic JSON samples
+
+Azure SDK client [protocol methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md) take `RequestContent` as an input parameter and return `Response` as their return type.  These types hold raw JSON content that can be accessed using Base Class Library (BCL) types such as `JsonDocument`, but use of these APIs can result in code that is difficult to read and obscures the author's intent.  To improve the developer experience, Azure.Core provides a [dynamic](https://learn.microsoft.com/dotnet/csharp/advanced-topics/interop/using-type-dynamic) layer over JSON APIs.
+
+## Accessing Response Content
+
+### Get `dynamic` from response
+
+Dynamic content is obtained from the `Response` return value.
+
+```C# Snippet:AzureCoreGetDynamicJson
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+```
+
+### Get a JSON property
+
+JSON members are read using dynamic property access.
+
+```C# Snippet:AzureCoreGetDynamicJsonProperty
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+string name = widget.Name;
+```
+
+You can learn what properties are available in the JSON response content from the REST API documentation for the service, examples in the protocol method documentation, or by expanding the [Dynamic View](https://learn.microsoft.com/visualstudio/debugger/watch-and-quickwatch-windows) in Visual Studio.
+
+If no parameter is passed to `ToDynamicFromJson()`, properties names must exactly match the member names in the JSON content.  Passing `DynamicDataOptions.Default` will enable properties to be accessed using PascalCase property names, and will write any added properties with camelCase names.
+
+### Set a JSON property
+
+JSON members can be set on the dynamic object.
+
+```C# Snippet:AzureCoreSetDynamicJsonProperty
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+widget.Name = "New Name";
+```
+
+### Access an array value
+
+JSON array values are accessed using array indexers.
+
+```C# Snippet:AzureCoreGetDynamicJsonArrayValue
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+
+// JSON is "{ values = [1, 2, 3] }"
+int value = widget.Values[0];
+```
+
+### Check whether an optional property is present
+
+Optional properties will return null if not present in the JSON content.
+
+```C# Snippet:AzureCoreGetDynamicJsonOptionalProperty
+Response response = await client.GetWidgetAsync("123");
```

Comment:
It would be good if it was clear what the payload looks like here, e.g. 
```csharp
Response response = await client.GetWidgetAsync(); // returns { "properties" : { "color" : "blue" }  }
```

Also, I find the "123" distracting. I would remove it. Maybe even "Async". All these things are not relevant to this guide.